decimal/bcd.go

```diff
@@ -185,10 +185,10 @@ func encodeStringToBCD(buf string) ([]byte, error) {
 // ended by a 4-bit sign nibble in the least significant four bits of the final
 // byte. The scale is used (negated) as the exponent of the decimal number.
 // Note that zeroes may have a sign and/or a scale.
-func decodeStringFromBCD(bcdBuf []byte) (string, error) {
+func decodeStringFromBCD(bcdBuf []byte) (string, int, error) {
 	// Index of a byte with scale.
 	const scaleIdx = 0
-	scale := int(bcdBuf[scaleIdx])
+	scale := int(int8(bcdBuf[scaleIdx]))
```

Comment:
Well, actually, it is not valid. You need to do honest messagepack decode here.

`scale` is either MP_INT or MP_UINT.

```lua
tarantool> msgpack.encode(decimal.new('1e33')):hex()
---
- c70301d0df1c
...
```
where `d0df` is a scale
```lua
tarantool> msgpack.decode(string.fromhex('d0df'))
---
- -33
- 3
...
```

Reply:
Thank you, updated:

https://github.com/tarantool/go-tarantool/compare/04f0f53ffaac403e5d1835518e58d392b71a1a3e..b67786fea2a52910bc32ba2ce996912a050ab546